### PR TITLE
Myceli integration tests

### PR DESCRIPTION
**Pull request by plauche**
### Added

- Added integration tests for `myceli` which get pretty close to end-to-end testing.
- Added helper functions to `Message` enum for generating specific messages.
- Implemented `TransmitBlock` API for transmitting a single block.

### Changed

- Reworked `myceli` crate a bit to expose internal library in binary crate.

### CODE REVIEW

File: myceli/tests/utils/mod.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        return TestListener {
+            listen_addr,
+            test_dir,
+        };
+    }
+
+    pub async fn start(&self) -> Result<()> {
+        let thread_listen_addr = self.listen_addr.to_owned();
+        let thread_db_path = self.test_dir.child("storage.db");
```
Comment (ddimaria) [inline myceli/tests/utils/mod.rs:36]: The names look the same, but not sure, would it be beneficial to use a separate DB for testing?

File: myceli/tests/listener_test.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    let response = controller
+        .send_and_recv(&listener.listen_addr, Message::request_available_blocks())
+        .await;
+
+    assert_eq!(response, Message::available_blocks(vec![]));
+}
+
+#[tokio::test]
+pub async fn test_transmit_receive_block() {
```
Comment (ddimaria) [inline myceli/tests/listener_test.rs:23]: There is some duplication in these tests.  Not a big deal, but there could be an abstraction to reuse for them.

File: myceli/tests/utils/mod.rs (hunk 1 above)
Comment (plauche) [reply myceli/tests/utils/mod.rs:36]: The generated `self.test_dir` is different temporary dir for each `TestListener`, so the DBs will end up as separate.

File: myceli/tests/listener_test.rs (hunk 2 above)
Comment (plauche) [reply myceli/tests/listener_test.rs:23]: Do you mean duplication in the setup for each test, or in the scenarios under test?

**Pull request merged**